src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -127,5 +128,19 @@ internal static MultiClusterRegistrationStrategy GetMultiClusterRegistrationStra
                             grainClass.Name));
             }
         }
+
+        // used by Orleankka's bootstrapper
+        internal void SetMessageAlwaysInterleaves(Func<object, bool> predicate)
+        {
+            if (predicate == null)
+                throw new ArgumentNullException(nameof(predicate));
+
+            IsMessageInterleavesPredicate = predicate;
+        }
+
+        public bool IsMessageInterleaves(Message message)
+        {
+            return IsReentrant || IsMessageInterleavesPredicate(message.BodyObject.GetType());
```

Comment:
I take it that in Orleankka you do not use the `InvokeMethodRequest` as the BodyObject, otherwise this wouldn't give you what you want